File: frontend/src/services/taskService.js
import axios from 'axios';

/**
 * Service for interacting with the tasks API
 */
const taskService = {
  /**
   * Get the most recent tasks
   * @returns {Promise} Promise containing task data
   */
  getRecentTasks: () => {
    return axios.get('/api/tasks');
  },

  /**
   * Create a new task
   * @param {Object} task - The task object
   * @param {string} task.title - The task title
   * @param {string} task.description - The task description
   * @returns {Promise} Promise containing the created task
   */
  createTask: (task) => {
    return axios.post('/api/tasks', task);
  },

  /**
   * Mark a task as complete
   * @param {number} id - The task ID
   * @returns {Promise} Promise containing the updated task
   */
  completeTask: (id) => {
    return axios.put(`/api/tasks/${id}/complete`);
  }
};

export default taskService;

File: backend/src/main/java/com/todo/task/backend/service/TaskService.java
package com.todo.task.backend.service;

import com.todo.task.backend.model.Task;

import java.util.List;

public interface TaskService {

    /**
     * Get the most recent incomplete tasks (up to 5)
     *
     * @return List of the most recent incomplete tasks
     */
    List<Task> getRecentTasks();

    /**
     * Create a new task
     *
     * @param task The task to create
     * @return The created task with generated ID
     */
    Task createTask(Task task);

    /**
     * Mark a task as complete
     *
     * @param id The task ID
     * @return The updated task
     */
    Task completeTask(Long id);
}